tests/generation/FileGlobs.py

```diff
@@ -71,12 +72,22 @@ def test_get_project_files_gitignore(self):
             "globexp.py",
             "upload.c",
             os.path.join("src", "main.pyc"),
+            os.path.join("another_folder", "script.js"),
             "run.pyc"]
 
         for file in files + ignored_files:
             os.makedirs(os.path.dirname(os.path.abspath(file)), exist_ok=True)
             open(file, "w").close()
         files += [".gitignore"]
+        os.chdir("another_folder")
+        with open('.gitignore', 'w') as f:
+            f.write("""
+# Start of gitignore
+*.js
+# End of gitignore""")
+        os.chdir(os.path.dirname(os.path.realpath(__file__)))
+        os.chdir("file_globs_gitignore_testfiles")
+        files += [os.path.join("another_folder", ".gitignore")]
```

Comment:
Not really required, but having 3 gitignores in different levels and then aggregating ignore globs from them would be more robust test case. Anyways, it's up to you